src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -40,27 +40,115 @@ object Homework :
 
   object `Boolean Operators` :
 
-    def not(b: Boolean): Boolean = ??? // here is my greatest solution
+    def not(b: Boolean): Boolean =
+      if (b == true)
+      then false
+      else true  // here is my greatest solution
 
-    def and(left: Boolean, right: Boolean): Boolean = ???
+    def and(left: Boolean, right: Boolean): Boolean =
+      if (left == true)
+      then right
+      else false
 
-    def or(left: Boolean, right: Boolean): Boolean = ???
+    def or(left: Boolean, right: Boolean): Boolean =
+      if (left == false)
+      then right
+      else true
 
   end `Boolean Operators`
 
   object `Fermat Numbers` :
 
-    val multiplication: (BigInt, BigInt) => BigInt = ???
+    val mult: (BigInt, BigInt, BigInt) => BigInt = (x, y, res) =>
+      if ((x<=0) || (y<=0))
+        then res
+      else mult(x-1, y, res+y)
 
-    val power: (BigInt, BigInt) => BigInt = ???
+    val multiplication: (BigInt, BigInt) => BigInt = (a, b) => mult(a,b,0)
 
-    val fermatNumber: Int => BigInt = ???
+    val pow: (BigInt, BigInt, BigInt) => BigInt = (aa, n, result) =>
+      if (n==0)
+        then result
+      else pow(aa, n-1, multiplication(aa, result))
+
+    val power: (BigInt, BigInt) => BigInt = (a, b) => pow(a, b, 1)
+
+    val fermatNumber: BigInt => BigInt = n => (power(2, power(2, n)) + 1)
+    println("fermatNumber(10)")
+    println(fermatNumber(10))
 
   end `Fermat Numbers`
 
   object `Look-and-say Sequence` :
-    val lookAndSaySequenceElement: Int => BigInt = ???
+    //val lookAndSaySequenceElement: BigInt => BigInt = ???
+
+    def xnext(res: BigInt, x: BigInt): BigInt =
+      if x == 0
```

Comment:
Please reformat the code. Curly braces should be on the same line with `if` or `else`.
Also if you use curly braces then `then` looks redundant.
Usually `then` is placed on the same line with `if`.